cases/migrations/0029_back_populate_case_last_update_type.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 3.2.15 on 2022-09-29 13:28
+# Empty migration filled in manually.
+
+from django.db import migrations
+from django.db.models import Max
+
+ACTION_TYPE = "AC"
+COMPLAINT_TYPE = "CO"
+
+
+def update_case(case):
+    result = case.complaints.aggregate(Max("created"))
+    latest_complaint_time = result["created__max"]
+
+    result = case.actions.aggregate(Max("time"))
+    lastest_action_time = result["time__max"]
+
+    if latest_complaint_time and lastest_action_time:
+        if lastest_action_time > latest_complaint_time:
+            case.last_update_type = ACTION_TYPE
+        else:
+            # Complaints win the tiebreaker.
+            case.last_update_type = COMPLAINT_TYPE
+    elif latest_complaint_time:
+        case.last_update_type = COMPLAINT_TYPE
+    elif lastest_action_time:
+        case.last_update_type = ACTION_TYPE
+    else:
+        # We somehow have a case without a complaint or actions:
+        # just ignore it.
+        return
+
+    case.save()
+
+
+def forwards_func(apps, schema_editor):
+    Case = apps.get_model("cases", "Case")
+
+    for case in Case.objects.all():
```

Comment:
I think you can use `Case.objects.annotate(Max("complaints__created"), Max("actions__created"))` here which would then give you all the info you want in one query (each case would have `actions__created__max` and `complaints__created__max` attributes).

Reply:
Nice!